newsroom/history_async.py

```diff
@@ -0,0 +1,178 @@
+import pymongo.errors
+from bson import ObjectId
+import werkzeug.exceptions
+
+from datetime import datetime
+from pydantic import BaseModel
+from typing import Optional, Dict, Annotated, List, Any, Union
+
+from newsroom import MONGO_PREFIX
+from newsroom.users.utils import get_user_or_abort
+from newsroom.utils import get_json_or_400
+from newsroom.core.resources.model import NewshubResourceModel
+from newsroom.core.resources.service import NewshubAsyncResourceService
+from newsroom.users.model import UserResourceModel
+
+from superdesk.core.module import Module
+from quart_babel import gettext
+from superdesk.utc import utcnow
+from superdesk.core.web import EndpointGroup, Response, Request
+from superdesk.flask import abort
+from superdesk.core import json
+from superdesk.core.resources.fields import ObjectId as ObjectIdField
+from eve.utils import ParsedRequest
+from superdesk.core.resources.validators import validate_data_relation_async
+from superdesk.core.resources import (
+    ResourceConfig,
+    MongoResourceConfig,
+    MongoIndexOptions,
+)
+
+
+class RouteParams(BaseModel):
+    item: Optional[Dict[str, Any]] = None
+    action: Optional[str] = None
+    section: Optional[str] = None
+
+
+class HistoryResourceModel(NewshubResourceModel):
+    action: Optional[str] = None
```

Comment:
`action`, `item` and `section` are marked as `not_analyzed` in the original schema.

For the new Elastic resources, we would use Keyword field instead of just string.

i.e.
```python
from superdesk.core.resources.fields import Keyword

class HistoryResourceModel(NewshubResourceModel):
    action: Keyword | None = None
    item: Keyword | None = None
    section: Keyword | None = None
```

Also please look at the current implementation, and determine if fields are optional or not.
For example, it looks like item action and section may be required fields